### ISSUE TITLE
`clang-analyzer-optin.cplusplus.UninitializedObject` false positive with unnamed fields

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#132001]
```cpp
struct S
{
    S(bool b)
    : b(b)
    {}
    bool b{false};
    long long : 7; // padding
};

void f()
{
    S s(true);
}
```

```
<source>:4:9: warning: 1 uninitialized field at the end of the constructor call [clang-analyzer-optin.cplusplus.UninitializedObject]
    4 |     : b(b)
      |         ^
<source>:7:15: note: uninitialized field 'this->'
    7 |     long long : 7; // padding
      |               ^
<source>:12:7: note: Calling constructor for 'S'
   12 |     S s(true);
      |       ^~~~~~~
<source>:4:9: note: 1 uninitialized field at the end of the constructor call
    4 |     : b(b)
      |         ^
```
https://godbolt.org/z/7zzoK97x5

**Comment by steakhal** [comment]
Probably somewhere in https://github.com/llvm/llvm-project/blob/main/clang/lib/StaticAnalyzer/Checkers/UninitializedObject/UninitializedObjectChecker.cpp we should check if the uninitialized field is an unnamed bitfield and bail out.

**Comment by YLChenZ** [comment]
@steakhal hey can you provide a bit more info on this issue and assign me I would like to work on it

**Comment by kr-2003** [comment]
@steakhal Here, are some of my findings and probable solution:
1. 
```
* frame #0: 0x00000001079a8db8 libclang-cpp.dylib`(anonymous namespace)::RegularField::printNoteMsg(llvm::raw_ostream&) const
    frame #1: 0x00000001079a8134 libclang-cpp.dylib`clang::ento::FieldChainInfo::printNoteMsg(llvm::raw_ostream&) const + 52
    frame #2: 0x00000001079a5e58 libclang-cpp.dylib`clang::ento::FindUninitializedFields::addFieldToUninits(clang::ento::FieldChainInfo, clang::ento::MemRegion const*) + 728
    frame #3: 0x00000001079a5658 libclang-cpp.dylib`clang::ento::FindUninitializedFields::isNonUnionUninit(clang::ento::TypedValueRegion const*, clang::ento::FieldChainInfo) + 968
    frame #4: 0x00000001079a51ec libclang-cpp.dylib`clang::ento::FindUninitializedFields::FindUninitializedFields(llvm::IntrusiveRefCntPtr<clang::ento::ProgramState const>, clang::ento::TypedValueRegion const*, clang::ento::UninitObjCheckerOptions const&) + 228
```
This is the backtrace while running ```lldb -- /opt/homebrew/opt/llvm/bin/clang -cc1 -analyze -analyzer-checker=optin.cplusplus.UninitializedObject test.cpp```. 
2. Execution goes from ```FindUninitializedFields::FindUninitializedFields``` constructor to ```FindUninitializedFields::addFieldToUninits``` function. 
3. Inside ```FindUninitializedFields::addFieldToUninits``` function, the it falls under this [block](https://github.com/llvm/llvm-project/blob/387f3e8f986d53067a68aa0d7b058a0ce81ec941/clang/lib/StaticAnalyzer/Checkers/UninitializedObject/UninitializedObjectChecker.cpp#L334):
```
    if (isPrimitiveType(T)) {
      if (isPrimitiveUninit(V)) {
        if (addFieldToUninits(LocalChain.add(RegularField(FR))))
          ContainsUninitField = true;
      }
      continue;
    }
```
4. Edit ```isNonUnionUninit``` (caller of ```isPrimitiveUninit```): Add a check before calling ```isPrimitiveUninit```
```
if (isPrimitiveType(T)) {
    // Skip unnamed bitfields (padding)
    if (I->isBitField() && I->isUnnamedBitfield()) {
        IsAnyFieldInitialized = true; // Treat as initialized to avoid warning
        continue;
    }
    if (isPrimitiveUninit(V)) {
        if (addFieldToUninits(LocalChain.add(RegularField(FR))))
            ContainsUninitField = true;
    }
    continue;
}
```

**Comment by YLChenZ** [comment]
@steakhal 
My solution is to directly determine if the current field declaration is an unnamed bitfield at the beginning of the [current loop](https://github.com/llvm/llvm-project/blob/387f3e8f986d53067a68aa0d7b058a0ce81ec941/clang/lib/StaticAnalyzer/Checkers/UninitializedObject/UninitializedObjectChecker.cpp#L293), like this 
```c++
for (const FieldDecl *I : RD->fields()) {
    if (I->isUnnamedBitField()) {
      IsAnyFieldInitialized = true;
      continue;
    }
    ......
}
```
At this point, we only care about whether the current field is unnamed or not, and do not go deeper to determine whether the type of the field is legal or not, because this has already been done in `isUnnamedBitField()`.

Here are my test results：

Total Discovered Tests: 989
  Unsupported      :  19 (1.92%)
  Passed           : 964 (97.47%)
  Expectedly Failed:   6 (0.61%)
